services/autoscaling/src/simcore_service_autoscaling/dynamic_scaling_core.py

```diff
@@ -107,13 +241,15 @@ async def check_dynamic_resources(app: FastAPI) -> None:
             )
 
             # NOTE: new_instance_dns_name is of type ip-123-23-23-3.ec2.internal and we need only the first part
-            if match := re.match(_EC2_INTERNAL_DNS_RE, new_instance_dns_name):
+            if match := re.match(
+                _EC2_INTERNAL_DNS_RE, new_instance_data.aws_private_dns
+            ):
                 new_instance_dns_name = match.group(1)
                 new_node = await utils_docker.wait_for_node(
-                    docker_client, new_instance_dns_name
+                    get_docker_client(app), new_instance_dns_name
                 )
                 await utils_docker.tag_node(
-                    docker_client,
+                    get_docker_client(app),
```

Comment:
if this call throws, is this handled in a nice way? I guess it could fail due to sheer statistical unluck, the docker swarm is a fragile system ;)

Reply:
same as above, catched by the periodic task handler. then restarted 5 seconds later.